runtime/src/main/java/org/corfudb/runtime/view/AbstractView.java

```diff
@@ -86,7 +89,13 @@ public Layout getCurrentLayout() {
         final Duration retryRate = runtime.getParameters().getConnectionRetryRate();
         while (true) {
             try {
-                return function.apply(new RuntimeLayout(runtime.layout.get(), runtime));
+                final Layout layout = runtime.layout.get();
```

Comment:
Place proper synchronization around this so only a single retrieval is permitted, use a stamped lock so that optimistic reads are possible.

Reply:
I used the atomic reference to provide synchronization. So that single retrieval is permitted.